core/src/main/scala/akka/persistence/jdbc/journal/dao/JournalQueries.scala

```diff
@@ -69,10 +69,10 @@ class JournalQueries(
     JournalTable.filter(_.persistenceId === persistenceId).filter(_.sequenceNumber <= toSequenceNr).delete
   }
 
-  def markJournalMessagesAsDeleted(persistenceId: String, maxSequenceNr: Long) =
+  def markAsDeleted(persistenceId: String, highestMarkedSequenceNr: Long) =
     JournalTable
       .filter(_.persistenceId === persistenceId)
-      .filter(_.sequenceNumber <= maxSequenceNr)
+      .filter(_.sequenceNumber === highestMarkedSequenceNr)
```

Comment:
being more explicitly and only marking a single message as delete. 

Only used when deleted events though and what we want is to delete the latest.

Reply:
Hmm, this will probably not work as expected. Looking further in the code.